core/embed/rust/src/ui/model_mercury/component/share_words.rs

```diff
@@ -18,11 +21,117 @@ use heapless::{String, Vec};
 
 const MAX_WORDS: usize = 33; // super-shamir has 33 words, all other have less
 const ANIMATION_DURATION_MS: Duration = Duration::from_millis(166);
+type IndexVec = Vec<usize, MAX_WORDS>;
```

Comment:
```suggestion
type IndexVec = Vec<u8, MAX_WORDS>;
```
I'm a fan of using correct int types but `usize` seems wasteful in such large array. Other possibility is to use some kind of bitvector.

Reply:
https://github.com/trezor/trezor-firmware/pull/3954/commits/94c4b7ff7b39a2bf103008c08f60248d17044fee